pkg/kubelet/qos/policy.go

```diff
@@ -62,13 +63,18 @@ func GetContainerOOMScoreAdjust(pod *v1.Pod, container *v1.Container, memoryCapa
 	// which use more than their request will have an OOM score of 1000 and will be prime
 	// targets for OOM kills.
 	// Note that this is a heuristic, it won't work if a container has many small processes.
-	memoryRequest := container.Resources.Requests.Memory().Value()
+	containerMemoryRequest := container.Resources.Requests.Memory().Value()
+	podRequests := resourcehelper.PodRequests(pod, resourcehelper.PodResourcesOptions{})
+	podMemoryRequest := podRequests.Memory().Value()
+
 	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
 		if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
-			memoryRequest = cs.AllocatedResources.Memory().Value()
+			containerMemoryRequest = cs.AllocatedResources.Memory().Value()
 		}
 	}
-	oomScoreAdjust := 1000 - (1000*memoryRequest)/memoryCapacity
+	// The OOM score is calculated based on an equal balance
+	// between the container request (50%) and the pod request (50%)
+	oomScoreAdjust := 1000 - (((500 * containerMemoryRequest) / memoryCapacity) + ((500 * podMemoryRequest) / memoryCapacity))
```

Comment:
Why did you choose this calculation logic? While there's a issue of how to set the threshold, I think it could have been done using the calculation logic suggested in Sergey's comment.

> Perhaps, use the entire pod's OOM score adjustment for the containers when those containers are requesting a fraction of memory of the entire pod's request.

Reply:
I concur that changing the logic deserves an explicit reference about the algorithm.
IIUC this change is needed by the sidecar containers, but it seems it can affects all the workloads, thus some extra care is warranted.

Reply:
In the case of a single container, containerMemoryRequest is approximately equal to podMemoryRequest (same logic).
For a multi-container pod, the OOM score adjustment is calculated as: 50% based on the container’s request + 50% based on the pod’s total request.
In my opinion, the new logic provides good behavior for both types of workloads (sidecar and non-sidecar), especially when distinguishing between larger and smaller pods.

I’m open to any feedback you may have.